lib/Controller/ImageController.php

```diff
@@ -0,0 +1,210 @@
+	 * @NoAdminRequired
+	 * @PublicPage
+	 *
+	 * @param int $documentId
+	 * @param int $sessionId
+	 * @param string $sessionToken
+	 * @param string $imagePath
+	 * @return DataResponse
+	 */
+	public function insertImageFile(int $documentId, int $sessionId, string $sessionToken, string $imagePath): DataResponse {
+		if (!$this->sessionService->isValidSession($documentId, $sessionId, $sessionToken)) {
+			return new DataResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+		$session = $this->sessionService->getSession($documentId, $sessionId, $sessionToken);
+		$userId = $session->getUserId();
+
+		try {
+			$insertResult = $this->imageService->insertImageFile($documentId, $imagePath, $userId);
+			return new DataResponse($insertResult);
+		} catch (Exception $e) {
+			$this->logger->error('File insertion error', ['exception' => $e]);
+			return new DataResponse(['error' => 'File insertion error'], Http::STATUS_BAD_REQUEST);
+		}
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @PublicPage
+	 *
+	 * @param string $link
+	 * @param int $documentId
+	 * @param int $sessionId
+	 * @param string $sessionToken
+	 * @param string|null $shareToken
+	 * @return DataResponse
+	 */
+	public function insertImageLink(string $link, int $documentId, int $sessionId, string $sessionToken, ?string $shareToken = null): DataResponse {
+		if (!$this->sessionService->isValidSession($documentId, $sessionId, $sessionToken)) {
+			return new DataResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+
+		try {
+			if ($shareToken) {
+				$downloadResult = $this->imageService->insertImageLinkPublic($documentId, $link, $shareToken);
+			} else {
+				$session = $this->sessionService->getSession($documentId, $sessionId, $sessionToken);
+				$userId = $session->getUserId();
+				$downloadResult = $this->imageService->insertImageLink($documentId, $link, $userId);
+			}
+			return new DataResponse($downloadResult);
+		} catch (Exception $e) {
+			$this->logger->error('Link insertion error', ['exception' => $e]);
+			return new DataResponse(['error' => 'Link insertion error'], Http::STATUS_BAD_REQUEST);
+		}
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @PublicPage
+	 *
+	 * @param int $documentId
+	 * @param int $sessionId
+	 * @param string $sessionToken
+	 * @param string|null $shareToken
+	 * @return DataResponse
+	 */
+	public function uploadImage(int $documentId, int $sessionId, string $sessionToken, ?string $shareToken = null): DataResponse {
+		if (!$this->sessionService->isValidSession($documentId, $sessionId, $sessionToken)) {
+			return new DataResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+
+		try {
+			$file = $this->request->getUploadedFile('image');
+			if ($file !== null && isset($file['tmp_name'], $file['name'], $file['type'])) {
+				if (!in_array($file['type'], self::IMAGE_MIME_TYPES, true)) {
+					return new DataResponse(['error' => 'Image type not supported'], Http::STATUS_BAD_REQUEST);
+				}
+				$newFileContent = file_get_contents($file['tmp_name']);
```

Comment:
Also no blocker but would be good to stream the file to the storage instead of loading it into memory. https://github.com/nextcloud/deck/blob/13dcacc3bb4149ebc7283f34f6d9af05322adea5/lib/Service/FilesAppService.php#L194-L198

Reply:
:+1: Done in d1cf1fb7f80b68bf3f3af0769c5e410ba831e2c2